include/data/json.h

```diff
@@ -37,8 +37,8 @@ enum json_tokens {
 	JSON_TOK_NONE = '_',
 	JSON_TOK_OBJECT_START = '{',
 	JSON_TOK_OBJECT_END = '}',
-	JSON_TOK_LIST_START = '[',
-	JSON_TOK_LIST_END = ']',
+	JSON_TOK_ARRAY_START = '[',
```

Comment:
If this is a public header, then it might be best to phase the old symbol out before removing it entirely.

Is it possible to deprecate the old enum and still add the new enum?

They can both refer to the same thing, in any case. The rest of this change looks fine to me.

Reply:
Agreed, we can leave both in the header before removing entirely. Is there a specific way I should mark that `JSON_TOK_LIST_START` and `JSON_TOK_LIST_END` will be deprecated in the future?

Reply:
@Sir-Branch - gcc will allow you to put arbitrary attributes after the identifier of an enum and before the comma. It does not create warnings, but if it compiles, I think it's fair game.

```cpp
enum foo {
  FOO,
  BAR
  __attribute__((deprecated("BAR will be removed use BAZ instead"))),
  BAZ,
};
```

Reply:
Awesome i'll try to get that out sometime tomorrow!

Reply:
As far as I can see, the members of this enum are only used internally, so I didn't think it was necessary to keep deprecated old versions. The actual API uses the correct term `ARRAY` already.

But it won't harm to deprecate them first and completely remove them in the future.

Reply:
@cfriedt - I went ahead and used the __deprecated zephyr define + an inline comment. My concern is that by using the __attribute__(....) from gcc might cause error with other compiler